paper.tex

```diff
@@ -329,6 +306,114 @@ \subsection{Unrolled recursion}
+
+The joins between adjacent segments need to form a continuous outline when combined with the segments. This requires that a thread have access to the tangent vector at the end of its assigned segment \emph{and} the start tangent of the next adjacent segment. We require that the input segments within a subpath are stored in order which simplifies the access to the adjacent segment down to a simple read at the next array offset. Co-operation across threads is not necessary.
+
+A property of this data-parallel structure is that the output is unordered. The cubic-Bézier-to-Euler-spiral conversion is sequential within a single thread, so it is possible to preserve ordering at the scope of an individual path segment, if desired. However, ordering cannot be guaranteed across path segments. We refer to this output data structure as \emph{line soup}. We believe this structure can be adapted to other primitives, such as circular arc segments or triangles. We incorporated our compute shader into our own massively-parallel GPU rasterizer that spatially sorts the line soup in subsequent binning and tiling stages. As such, maintaining any sort of ordering with respect to the input path data structure was not necessary.
+
+% TODO: Talk about the inner join here. We detect inner vs outer outline based on join tangent cross product. We just draw a straight line for the inner join.
+% TODO: line soup is bump allocated. Mention this here? I'm leaning towards discussing that in the Limitations section below, which will focus on memory allocation and estimation.
+% TODO: while parallel on input, sequential per curve. However iterations are bounded (? needs measurement). Should this discussion happen here or down in the Performance section?
+
+
+\subsection{Input encoding}
+
+% TODO: not sure how much to emphasize the path tag encoding. The input processing monoids benefit the vello scene API (fragments) the TagMonoid could generally be computed on the CPU (with minimal offset tracking) It's not that relevant to the stroke processing. The main bit to mention is that it's a struct-of-arrays.
```

Comment:
We can skip over details of the encoding, but I do think the stream of tag bytes + stream of coordinates part is important, as it's central to the compact encoding.